MaterialLib/MPL/Properties/CreateCurveProperty.cpp

```diff
@@ -0,0 +1,52 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2020, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreateCurveProperty.h"
+
+#include "BaseLib/ConfigTree.h"
+#include "CurveProperty.h"
+#include "MaterialLib/MPL/VariableType.h"
+#include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
+
+namespace MaterialPropertyLib
+{
+std::unique_ptr<CurveProperty> createCurveProperty(
+    BaseLib::ConfigTree const& config,
+    std::map<std::string,
+             std::unique_ptr<MathLib::PiecewiseLinearInterpolation>> const&
+        curves)
+{
+    DBUG("Create CurveProperty.");
+    //! \ogs_file_param{properties__property__type}
+    config.checkConfigParameter("type", "Curve");
+
+    //! \ogs_file_param{properties__property__Curve__curve}
+    auto curve_name = config.getConfigParameter<std::string>("curve");
+    DBUG("Using curve '%s'", curve_name.c_str());
+
+    auto const curve_it = curves.find(curve_name);
+    if (curve_it == curves.end())
+    {
+        OGS_FATAL("Curve '%s' does not exists.", curve_name.c_str());
+    }
```

Comment:
```suggestion
    auto const& curve = *BaseLib::getOrError(curves, curve_name, "Could not find curve.");
```

Reply:
OGS_FATAL("Curve '%s' does not exist.", curve_name.c_str());

Reply:
@Scinopode This is already included in the [`getOrError`](https://github.com/ufz/ogs/blob/master/BaseLib/Algorithm.h#L147-L159) function.